tests/unit/processor/template_replacer/test_template_replacer.py

```diff
@@ -155,4 +159,9 @@ def test_replace_fails_with_invalid_template(self):
             {"template": "tests/testdata/unit/template_replacer/replacer_template_invalid.yml"}
         )
         with pytest.raises(TemplateReplacerError, match="Not enough delimiters"):
-            Factory.create({"test instance": config}, self.logger)
+            self._create_template_replacer(config)
+
+    def _create_template_replacer(self, config):
+        template_replacer = Factory.create({"test instance": config}, self.logger)
+        template_replacer.setup()
+        return template_replacer
```

Comment:
consider to use a pytest fixture for this:


```suggestion
    @pytest.fixture(name="template_replacer")
    def _create_template_replacer(self, config):
        template_replacer = Factory.create({"test instance": config}, self.logger)
        template_replacer.setup()
        return template_replacer
```

and use it in the test methods parameter under its name.

Reply:
I'm not sure how to use this, since it requires a parameter from the test.

I could do something like this:
```python
@pytest.fixture(name="template_replacer")
def fixture_create_template_replacer(self):
    def _create_template_replacer(config):
        template_replacer = Factory.create({"test instance": config}, self.logger)
        template_replacer.setup()
        return template_replacer
    return _create_template_replacer
```